web/src/main/kotlin/com/example/ajaxproject/nats/UserUpdatedNatsServiceImpl.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.ajaxproject.nats
+
+import com.example.ajax.User
+import com.example.ajaxproject.KafkaTopic
+import com.google.protobuf.Parser
+import com.pubsub.user.UserUpdatedEvent
+import io.nats.client.Connection
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+
+@Service
+class UserUpdatedNatsServiceImpl(
+    private val connection: Connection
+) : UserEventNatsService<UserUpdatedEvent> {
+
+    override val parser: Parser<UserUpdatedEvent> = UserUpdatedEvent.parser()
+
+    private val dispatcher = connection.createDispatcher()
+
+    override fun subscribeToEvents(userId: String, eventType: String): Flux<UserUpdatedEvent> =
+        Flux.create { sink ->
+            dispatcher.apply {
+                subscribe(  "USER_PREFIX.$userId.$eventType")
```

Comment:
consider moving subject creation to NatsSubject

Reply:
I did another refactoring - create a function to create subject / topic which i need

Reply:
Also , refactored proto file to not write KafkaTopic each time